File: packages/python/src/qagentic/core/status.py
"""Test status definitions."""

from enum import Enum


class Status(str, Enum):
    """Test execution status."""
    
    PASSED = "passed"
    FAILED = "failed"
    BROKEN = "broken"
    SKIPPED = "skipped"
    PENDING = "pending"
    RUNNING = "running"
    UNKNOWN = "unknown"
    
    def __str__(self) -> str:
        return self.value
    
    @property
    def is_successful(self) -> bool:
        """Check if status represents a successful outcome."""
        return self == Status.PASSED
    
    @property
    def is_failure(self) -> bool:
        """Check if status represents a failure."""
        return self in (Status.FAILED, Status.BROKEN)
    
    @classmethod
    def from_pytest_outcome(cls, outcome: str) -> "Status":
        """Convert pytest outcome to Status."""
        mapping = {
            "passed": cls.PASSED,
            "failed": cls.FAILED,
            "skipped": cls.SKIPPED,
            "error": cls.BROKEN,
            "xfailed": cls.PASSED,  # Expected failure
            "xpassed": cls.FAILED,  # Unexpected pass
        }
        return mapping.get(outcome.lower(), cls.UNKNOWN)

File: packages/javascript/src/core/status.ts
/**
 * Test execution status.
 */

export enum Status {
  PASSED = 'passed',
  FAILED = 'failed',
  BROKEN = 'broken',
  SKIPPED = 'skipped',
  PENDING = 'pending',
  RUNNING = 'running',
  UNKNOWN = 'unknown',
}

/**
 * Check if status represents a successful outcome.
 */
export function isSuccessful(status: Status): boolean {
  return status === Status.PASSED;
}

/**
 * Check if status represents a failure.
 */
export function isFailure(status: Status): boolean {
  return status === Status.FAILED || status === Status.BROKEN;
}

/**
 * Parse status from string.
 */
export function parseStatus(value: string): Status {
  const normalized = value.toLowerCase();
  switch (normalized) {
    case 'passed':
    case 'pass':
    case 'success':
      return Status.PASSED;
    case 'failed':
    case 'fail':
    case 'failure':
      return Status.FAILED;
    case 'broken':
    case 'error':
      return Status.BROKEN;
    case 'skipped':
    case 'skip':
    case 'pending':
      return Status.SKIPPED;
    case 'running':
      return Status.RUNNING;
    default:
      return Status.UNKNOWN;
  }
}
